### DISCUSSION TITLE
sqlalchemy paginate is not returning the expected number of records

### DISCUSSION THREAD

**Discussion by ktriviso** [sqlalchemy/sqlalchemy#11765] [Usage Questions]
I have a query with two joins. Prior to using `.paginate`, I am able to see all the correct 48 records. After running `.paginate` it returns significantly less than expected and there are no "next pages" we need to parse. If I comment out the second join (OrderProducts), the issue is fixed and `.paginate` returns the expected records again.

Does anyone know how to fix this issue so that keeping both joins will return the expected 48 records?

`query = db.session.query(Order).join(Subscriber, Order.subscriber_id==Subscriber.subscriber_id).join(OrderProducts, Order.order_id == OrderProducts.order_id).order_by(Order.created_at)`
`print(query.all()) # returns 48 results`
`query = query.paginate(page=1, per_page=20, error_out=False)`
`print(query.total) # returns 6 results`

**Comment by zzzeek** [comment]
hi -

"paginate" is not part of SQLAlchemy, this might be part of flask-SQLAlchemy?   [looks like it is](https://flask-sqlalchemy.palletsprojects.com/en/3.0.x/pagination/) so I'd check with their tracker for this

though I will give a hint that if this paginate routine is using LIMIT/OFFSET, it might have issues with JOINs since these change the number of rows per primary record.

**Reply by ktriviso** [reply]
Thank you so much. It looks like they have a discord I can reach out on. Thank you!

**Reply by CaselIT** [reply]
it's likely that when using joins you have to apply the limit and offset to a subquery then join between the subquery and the other tables.
